docs/gettingstarted.rst

```diff
@@ -229,7 +229,81 @@ a string or integer. Below is an implementation of this design in the Python cod
++-------------+--------------------------------------------+-----------------------------------------------------------------------------------------------------+
+| type        | int                                        | An `ApplicationCommandPermissionType`_.                                                             |
++-------------+--------------------------------------------+-----------------------------------------------------------------------------------------------------+
+| permission  | boolean                                    | ``true`` to allow, ``false`` to disallow.                                                           |
++-------------+--------------------------------------------+-----------------------------------------------------------------------------------------------------+
+
+How the type parameter works is very simple. Discord has many ids to represent different things. As you can 
+set permissions to apply for User or Role, `ApplicationCommandPermissionType`_ is used. It's a number and
+following table shows the supported id types for permissions:
+
++-------------------+-----------+
+| **Name**          | **Value** |
++-------------------+-----------+
+| Role              | 1         |
++-------------------+-----------+
+| User              | 2         |
++-------------------+-----------+
+
+This is an example of how a single permission will look when represented as a json object:
+
+.. code-block:: python
+
+  {
+    "id": 12345678,
+    "type": 1,
+    "permission": True 
+  }
+
+Now, let take a look simple example. The slash command decorator have a permissions parameter Where
+it takes in a dictionary. The key being the guild to apply permissions on, and value being the list
+of permissions to apply. For each permission, we can use the handy ``create_permission`` method to 
+build the permission json explain above.
+
+In this case, we are setting 2 permissions for guild with id of ``12345678``. Firstly, we are allowing
+role with id ``99999999`` and disallowing user with id ``88888888`` from running the slash command.
+
+.. code-block:: python
+
+  from discord_slash.utils.manage_commands import create_permission
+
+  @slash.slash(name="test",
+              description="This is just a test command, nothing more.",
+              permissions={
+                12345678: [
+                  create_permission(99999999, 1, True),
+                  create_permission(88888888, 2, False)
+                ]
+             })
+  async def test(ctx):
+    await ctx.send(content="Hello World!")
+
+Additionally, we could also declare the id type of our permission through this method shown here:
+
+.. code-block:: python
+
+  from discord_slash.model import SubcommandApplicationPermissionType
+  
+  (...)
+  
+                  option_type=SubcommandApplicationPermissionType.ROLE
```

Comment:
How about just using the enum in the example, I think using the enums in files is always a good idea as it prevents confusion.

Reply:
sure